kafka-0.8.x/src/main/scala/monix/kafka/Serializer.scala

```diff
@@ -27,14 +27,18 @@ import language.existentials
   */
 final case class Serializer[A](
   className: String,
-  classType: Class[_ <: KafkaSerializer[A]]) {
+  classType: Class[_ <: KafkaSerializer[A]],
+  classInstance: Serializer.ClassInstanceProvider[A] = (s: Serializer[A]) => s.classType.newInstance()) {
```

Comment:
Default parameter looks ugly (personal pet peeve). I'm fairly certain that Scala accepts this:

```scala
classInstance: Serializer.ClassInstanceProvider[A] = _.classType.newInstance()
```

While we are at it `classInstance: Serializer.ClassInstanceProvider` is a little redundant and verbose. So lets simplify to:

```scala
constructor: Serializer.Constructor[A] = _.classType.newInstance()

// ... 

object Serializer {
  type Constructor[A] = (Serializer[A]) => KafkaSerializer[A]
  // ...
```

Naming, right? :-) I don't care that much, so it's up to you.